### ISSUE TITLE
Metrc Interface and ReplaceFieldValue function

### ISSUE THREAD

**Issue by Alessar** [influxdata/telegraf#3291]
### Feature Request
`Metric` interface contains `AddField`and `RemoveField` functions.
It will be good if this interface has function `ReplaceField` for replace field value.

**Comment by danielnelson** [comment]
Of course you can remove and then add, can you explain what your use case is which makes this a particular problem?

**Comment by Alessar** [comment]
For example, I need change metric value. So I do it and check results:
```
RemoveField(name)
AddField(name, value)
for <field in fields> {
  Printf(field)
}
```
Unfortunately, exeption happens in loop for.

**Comment by danielnelson** [comment]
Can you report that as a bug with more details about setup of the metric and the actual values?

Can you tell me a little bit about why you are using the these functions?  Are you writing a processor?

**Comment by Alessar** [comment]
It happens on any metric with more than one values.
It happens on any values, so my assumption that some bug is situated in internal code, which parses values.
Yes, task is writing processor, which can change values, thats why I need ReplaceFieldValue function.

**Comment by danielnelson** [comment]
I am evaluating some changes to the Metrics implementation, and I'll make sure to fix this issue as part of it.  The way this should work is that when you call `AddField`, if the field exists it should be replaced.

**Comment by Alessar** [comment]
I don't know behavior after your changes, which you want make,
but some time ago after `AddField` I could see unfortunately two fields with the same name.
So thats why I imagined that `ReplaceFieldValue` can be good solution for my case.

**Comment by danielnelson** [comment]
@Alessar The `AddField` function should work as expected now, the fix will be in 1.6
